extensions/amp-analytics/0.1/test/test-variables.js

```diff
@@ -59,27 +61,33 @@ describes.fakeWin('amp-analytics.VariableService', {amp: true}, env => {
     };
 
     function check(template, expected, vars) {
-      const actual = variables.expandTemplateSync(
+      const actual = variables.expandTemplate(
         template,
-        new ExpansionOptions(vars)
+        new ExpansionOptions(vars),
+        fakeElement
       );
-      expect(actual).to.equal(expected);
+      return expect(actual).to.eventually.equal(expected);
     }
 
     it('expands nested vars (encode once)', () => {
-      check('${a}', 'https%3A%2F%2Fwww.google.com%2Fa%3Fb%3D1%26c%3D2', vars);
+      return check(
```

Comment:
Are there test coverage for expanding the nested macro? Please help link to the test. Thanks. 
Also given there are a bunch of special cases, it would be great to add a manual test file which helps demonstrate all the expected behaviors. Thanks

Reply:
Added a manual file, let me know if there is anything else that you can think of. There is a test here  https://github.com/ampproject/amphtml/blob/d16512018ef4697cd47d0f85e440e3937c0174ce/extensions/amp-analytics/0.1/test/test-variables.js#L148-L153
that shows the new behavior, let me know if have idea for more.

Reply:
LG